### ISSUE TITLE
users are able to use the latest wing sdk version and run the console

### ISSUE THREAD

**Issue by ainvoner** [winglang/wing#1583]
### Community Note

> Please vote by adding a 👍 reaction to the issue to help us prioritize.
> If you are interested to work on this issue, please leave a comment.

### Feature Spec

Users are able to write their wing application using the latest wing-sdk version and run the Wing Console without worrying about SDK versions compatibility.

### Use Cases

Develop a Wing application using the latest wing version while running an older version of the Wing Console.
Since each Wing Console release is using a fixed version of the SDK to instantiate the local simulator, there might be cases were the globally installed wing sdk version is different than the sdk version bundled in the Console.

### Implementation Notes

For initial phase (P1 ?)

1. When compiling Wing application to a local target, add the sdk version to the simulator.json - already implemented
2. Each non breaking wing version release will trigger the Console 'dependencies upgrade' automatic workflow. For non breaking changes. automatically merge this PR to the main branch and release a new Console app.
3. The Console will try to instantiate the simulator, if instantiation will fail with 'incompatible version error' - the Console will display a warning to the user with a manual update button to update the console to the latest release
4. Add manual updates support to the Console.

### Component

SDK, Wing Console

**Comment by Chriscbr** [comment]
@ainvoner If someone is adding a new feature to the Wing compiler (that doesn't affect one of the main console APIs) how can they run a Wing application in the Wing console using their locally-built version of Wing? How is the Wing console finding the `wing` command to use to compile programs etc.? We would like to add some documentation about this for contributors.

cc @hasanaburayyan you showed me some issues where you tried using your locally built version of Wing but it didn't seem like the Wing console picked it up... when you have a chance could you provide steps to repro this? (If it's still an issue)

**Comment by hasanaburayyan** [comment]
Here is what I attempted but it was not successful

1. I updated my path to be prefixed with my local build bin. 
2. Updated something silly in the sdk
```js
public async put(key: string, value: string): Promise<void> {
    return this.context.withTrace({
      message: `Putzzzzzzz (key=${key}).`,      <==== I really like z's
      activity: async () => {
        return this.addFile(key, value);
      },
    });
  }
```
3. Built my local version of wing
4. Ran console and can see from gui the correct version is selected but logs are not matching
<img width="417" alt="image" src="https://user-images.githubusercontent.com/45375125/224838752-2b7037ec-5991-45f1-81ac-995581f788dc.png">
